core/local/local.go

```diff
@@ -346,7 +348,13 @@ func (e *ExecutionScheduler) Run(
 	executorsCount := len(e.executors)
 	logger := e.logger.WithField("phase", "local-execution-scheduler-run")
 	e.initProgress.Modify(pb.WithConstLeft("Run"))
-	defer e.state.MarkEnded()
+	var interrupted bool
+	defer func() {
+		e.state.MarkEnded()
+		if interrupted {
+			e.state.SetExecutionStatus(lib.ExecutionStatusInterrupted)
+		}
```

Comment:
can we get a comment that we on purpose first hit the `ended` and then the `interrupted` state

Reply:
:man_shrugging: Not sure, doesn't the code speak for itself?